### ISSUE TITLE
How to finetune with the VGG_coco_SSD_500x500_iter_200000.caffemodel ？

### ISSUE THREAD

**Issue by yanxp** [weiliu89/caffe#281]
I miss the problem Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_loc'; shape mismatch. Source param shape is 16 512 3 3 (73728); target param shape is 12 512 3 3 (55296). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.?

**Comment by chienyiwang** [comment]
You can rename the mbox layer in network by providing the postfix parameter (already implemented in the function)

**Comment by yanxp** [comment]
@chienyiwang  which function ? thank you !

**Comment by chienyiwang** [comment]
For example, in "ssd_pascal.py" script, the mbox_layers creation:
mbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,
        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,
        aspect_ratios=aspect_ratios, normalizations=normalizations,
        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,
        prior_variance=prior_variance, kernel_size=3, pad=1)
you can pass in "conf_postfix" and "loc_postfix" in the end to change the name of those layers:
mbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,
        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,
        aspect_ratios=aspect_ratios, normalizations=normalizations,
        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,
        prior_variance=prior_variance, kernel_size=3, pad=1, conf_postfix='_my_layer', loc_postfix='_my_layer')

**Comment by yanxp** [comment]
thank you very much!i来自我的华为手机-------- 原始邮件 --------主题：Re: [weiliu89/caffe] How to finetune with the VGG_coco_SSD_500x500_iter_200000.caffemodel ？ (#281)发件人：Chien-Yi Wang 收件人：weiliu89/caffe 抄送：yanxp ,Author For example, in "ssd_pascal.py" script, the mbox_layers creation:
mbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,
        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,
        aspect_ratios=aspect_ratios, normalizations=normalizations,
        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,
        prior_variance=prior_variance, kernel_size=3, pad=1)
you can pass in "conf_postfix" and "loc_postfix" in the end to change the name of those layers:
mbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,
        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,
        aspect_ratios=aspect_ratios, normalizations=normalizations,
        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,
        prior_variance=prior_variance, kernel_size=3, pad=1, conf_postfix='_my_layer', loc_postfix='_my_layer')

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/weiliu89/caffe","title":"weiliu89/caffe","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/weiliu89/caffe"}},"updates":{"snippets":[{"icon":"PERSON","message":"@chienyiwang in #281: For example, in \"ssd_pascal.py\" script, the mbox_layers creation:\r\nmbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,\r\n        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,\r\n        aspect_ratios=aspect_ratios, normalizations=normalizations,\r\n        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,\r\n        prior_variance=prior_variance, kernel_size=3, pad=1)\r\nyou can pass in \"conf_postfix\" and \"loc_postfix\" in the end to change the name of those layers:\r\nmbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,\r\n        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,\r\n        aspect_ratios=aspect_ratios, normalizations=normalizations,\r\n        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,\r\n        prior_variance=prior_variance, kernel_size=3, pad=1, conf_postfix='_my_layer', loc_postfix='_my_layer')"}],"action":{"name":"View Issue","url":"https://github.com/weiliu89/caffe/issues/281#issuecomment-260422184"}}}

**Comment by yanxp** [comment]
Thank you very much !

**Comment by zgplvyou** [comment]
@chienyiwang I got the same error with him. I modified "video_ssd_pascal.py" script as your solution. But the output is strange, there are so many bounding boxes (labed in 'train' ' person' ‘shape’ ‘cat’ ...VOC_20_class)in the video.I trained the model with own data just for detecting person , so it should have appeared only two classes(person and background) in the output. I didn't modify the train.prototxt and test.prototxt before training the model.

**Comment by Anhaoxu** [comment]
@yanxp @chienyiwang 
I use this method,but too many bbox appeared, and they are wrong,any suggestions?

**Comment by rain2008204** [comment]
@zgplvyou I have the same problems, so, how did you solve this? thank you very much.

**Comment by unanan** [comment]
https://stackoverflow.com/questions/39811936/using-bvlc-googlenet-as-pretrained-model-in-digits-errors/39837047#39837047
This SO page may help?

> When you rename the layer, Caffe doesn't try to copy the weights for that layer and you get randomly initialized weights
